frontend/src/console/components/InstanceSettings/InstanceSettings.tsx

```diff
@@ -66,6 +69,24 @@ export class InstanceSettingsComponent extends React.PureComponent<
         publicContactEmail: this.props.activeInstance!.data!.publicContactEmail
       });
     }
+    // if current state is empty and pre-filling
+    if (
+      (this.state.instanceName === '' ||
+        this.state.publicContactEmail === '') &&
+      prevProps.activeInstance.data &&
+      this.props.activeInstance.data &&
+      this.state.emptyFields === '1'
+    ) {
+      this.setState({
+        instanceName:
+          prevProps.activeInstance.data.instanceName ||
+          this.props.activeInstance!.data!.instanceName,
+        publicContactEmail:
+          prevProps.activeInstance.data.publicContactEmail ||
+          this.props.activeInstance!.data!.publicContactEmail,
+        emptyFields: '0'
+      });
+    }
```

Comment:
Unfortunately it seems to be refreshing the content only after receiving the response from `api/v2/instances/openedx_deployment/1/`. However the required data is already available, as it's returned by `api/v2/instances/openedx_config/`.
Please see the recording below:

![settings](https://user-images.githubusercontent.com/3189670/81332926-16d02f00-90a4-11ea-9677-9cd7f86ba248.gif)

The gif below represents what happens when you disable Ocim after the initial page load and before receiving the response to `api/v2/instances/openedx_config/`. The `InstanceSettingsComponent` already has the instance name and contact email in place:
![settings2](https://user-images.githubusercontent.com/3189670/81334583-90691c80-90a6-11ea-9582-84161b787d5b.gif)

Reply:
@Agrendalath Thanks for this thorough explanation, I have updated it now, whenever the component receiving the props I am updating it right away now :+1: